### ISSUE TITLE
Bug in multiprecision shift left

### ISSUE THREAD

**Issue by chfast** [boostorg/multiprecision#31]
In cpp_int this test fails:

```
using uint256 = number<cpp_int_backend<256, 256, unsigned_magnitude, unchecked, void>>;
BOOST_CHECK_EQUAL(uint256(3) << 255, uint256(1) << 255);
```

**Comment by jzmaddock** [comment]
Works for me with both msvc and gcc and current develop (and last release).  I *think* I remember fixing something like this, but can't find the commit at present.... please reopen if you still have issues with Boost-1.66.

**Comment by chfast** [comment]
Thanks. I will check 1.66.

**Comment by chfast** [comment]
I confirm this is fixed in 1.65.1 (the version we are currently using).
